Pal 883 examples documentation (#88)
gchq/Palisade-examples: README.md

-->
# Examples

> The example demonstrates different users querying an avro file over a REST api. These examples are run in 3 different scenarios as described in the [Deployment](deployment) and [Deployment-jvm](deployment-jvm) modules:

Reviewer: Prefer line-breaks in markdown every sentence
```suggestion
The example demonstrates different users querying an avro file over a REST api.
These examples are run in 3 different scenarios as described in the [Deployment](deployment) and [Deployment-jvm](deployment-jvm) modules:
```

Author: Updated

- [local-jvm](./deployment-jvm/local-jvm/README.md)
- [local-k8s](./deployment/local-k8s/README.md)
- [aws-k8s](./deployment/aws-k8s/README.md)

> The rest of the modules contain all the necessary classes and objects required for running the examples in those environment

Author: Updated

> The rest of the modules contain all the necessary classes and objects required for running the examples in those environment

Author: Done

The Palisade-examples repository contains all the example specific modules as follows:

### [Deployment](deployment)

> Contains the deployment specific code and scripts for running the example within a Kubernetes environment. Current deployment targets are:

Author: Updated

* [AWS Docker/Kubernetes Containers](./deployment/aws-k8s/README.md)
* [Local Docker/Kubernetes Containers](./deployment/local-k8s/README.md)

### [Deployment-jvm](deployment-jvm)

> Contains the deployment specific code and scripts for running the example within a local JVM environment. This can only be run locally

Reviewer: ```suggestion
Contains the deployment specific code and scripts for running the example within a local JVM environment.
These are intended to be run locally, but there are provided dockerfiles and charts to run in K8s.
```

* [Local JVM Processes](./deployment-jvm/local-jvm/README.md)

### [Example Library](example-library/README.md)

> The Example Library is a collection of Java classes that are required for running the example. For example, the `ExampleUser` class, as well as a number of fine-grained and coarse-grained policies that will be applied to the returned records.

Reviewer: ```suggestion
The Example Library is a collection of Java classes that are required for running the example.
For example, the `ExampleUser` class, as well as a number of fine-grained and coarse-grained policies that will be applied to the returned records.
```
There's quite a few of these, I'll leave you to the rest

The `ExampleUser` datatype is a specialisation of the [Palisade-common](https://github.com/gchq/Palisade-common) `User` class and contains specific extra fields. 

> The policies applied when running the example are a collection of static coarse-grained (resource-level) and fine-grained (record-level) rules that can be found in the *Example Library*, and are used for pre-populating the palisade services with data.

Author: Updated

> The Examples module contains all example specific modules as follows:

Author: The first paragraph has been re-worded.

> ### [Deployment](deployment)

Author: This is the name of the module so I would be inclined to leave it until it changes

> ### [Deployment](deployment)

Reviewer 2: Ok, but it could do with something that makes it clear that there are three deployments.

Contains the deployment specific code and scripts for running the example within a Kubernetes environment. Current deployment targets are:
* [AWS Docker/Kubernetes Containers](./deployment/aws-k8s/README.md)
* [Local Docker/Kubernetes Containers](./deployment/local-k8s/README.md)

### [Deployment-jvm](deployment-jvm)
Contains the deployment specific code and scripts for running the example within a locl JVM environment. This can only be run locally
* [Local JVM Processes](./deployment-jvm/local-jvm/README.md)

> 

Author: A new 1st sentence has been added

### [Example Library](example-library/README.md)
The policies applied when running the example are a collection of static coarse-grained (resource-level) and fine-grained (record-level) rules that can be found in the *Example Library*, and are used for prepopulating the palisade services with data.

> This collection of example-specific rules, types and configurations is based around the possible policies a company might set out for users accessing sensitive employee data, depending upon their role in the company.

Author: Added an extra line at the start

The performance of the palisade services compared to native file reads (and other metrics) can be measured using the *Performance* testing suite.
This performance-testing suite has a number of different trial types, datasets and policy variants to cover a reasonable number of common use-cases.

> Since scripts are all `.sh` files, the [services-manager](https://github.com/gchq/Palisade-services/blob/develop/services-manager/README.md) may be preferred for cross-platform compatibility.

Author: I've removed this as I'm not sure it's needed
